cli/command_repository_upgrade_test.go

```diff
@@ -0,0 +1,205 @@
+package cli_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/kopia/kopia/repo/content"
+	"github.com/kopia/kopia/tests/testenv"
+)
+
+func (s *formatSpecificTestSuite) TestRepositoryUpgrade(t *testing.T) {
+	env := testenv.NewCLITest(t, s.formatFlags, testenv.NewInProcRunner(t))
+
+	env.RunAndExpectSuccess(t, "repo", "create", "filesystem", "--path", env.RepoDir)
+	out := env.RunAndExpectSuccess(t, "repository", "status", "--upgrade-no-block")
+
+	switch s.formatVersion {
+	case content.FormatVersion1:
+		require.Contains(t, out, "Format version:      1")
+		_, stderr := env.RunAndExpectSuccessWithErrOut(t, "repository", "upgrade",
+			"--upgrade-owner-id", "owner",
+			"--io-drain-timeout", "1s",
+			"--status-poll-interval", "1s",
+			"--max-clock-drift", "1s")
+		require.Contains(t, stderr, "Repository indices have been upgraded.")
+		require.Contains(t, stderr, "Repository has been successfully upgraded.")
+	case content.FormatVersion2:
+		require.Contains(t, out, "Format version:      2")
+		_, stderr := env.RunAndExpectSuccessWithErrOut(t, "repository", "upgrade",
+			"--upgrade-owner-id", "owner",
+			"--io-drain-timeout", "1s",
+			"--status-poll-interval", "1s",
+			"--max-clock-drift", "1s")
+		require.Contains(t, stderr, "Repository indices have already been migrated to the epoch format, no need to drain other clients")
+		require.Contains(t, stderr, "Repository has been successfully upgraded.")
+	default:
```

Comment:
Just thinking about the usage here: the "upgrade" API is not idempotent (i.e. two subsequent upgrade commands won't give the same result). I'm sure if we publish documentation it'll be okay, but I wonder if there's any potential for trouble.